datumaro/plugins/data_formats/datumaro_binary/mapper/media.py

```diff
@@ -0,0 +1,113 @@
+# Copyright (C) 2023 Intel Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import struct
+from typing import Optional, Tuple
+
+from datumaro.components.errors import DatumaroError
+from datumaro.components.media import Image, MediaElement, MediaType, PointCloud
+
+from .common import Mapper, StringMapper
+
+
+class MediaMapper(Mapper):
+    @classmethod
+    def forward(cls, obj: Optional[MediaElement]) -> bytes:
+        if obj is None:
+            return struct.pack(f"<I", MediaType.NONE)
+        elif obj.MEDIA_TYPE == MediaType.IMAGE:
+            return ImageMapper.forward(obj)
+        elif obj.MEDIA_TYPE == MediaType.POINT_CLOUD:
+            return PointCloudMapper.forward(obj)
+        elif obj.MEDIA_TYPE == MediaType.MEDIA_ELEMENT:
+            return MediaElementMapper.forward(obj)
+        else:
+            raise DatumaroError(f"{obj.MEDIA_TYPE} is not allowed for MediaMapper.")
```

Comment:
Could you expect the operation when path is too long to convert string to 8 byte?